app/assets/components/messages/src/index.ts

```diff
@@ -0,0 +1,25 @@
+import App from './main';
+import { ComponentManager, setPropertyDidChange } from '@glimmer/component';
+import initializeCustomElements from '@glimmer/web-component';
+
+declare const ADVOCATE;
+
+const app = new App();
+
+setPropertyDidChange(() => {
+  app.scheduleRerender();
+});
+
+app.registerInitializer({
+  initialize(registry) {
+    registry.register(`component-manager:/${app.rootName}/component-managers/main`, ComponentManager)
+  }
+});
+
+// GLOBAL from application.js
+ADVOCATE.document_ready(() => {
```

Comment:
I customize this so that the `div` will be turned into the component regardless of where when we load javascript (head vs. end-of-body).